Navigation Link: Preserve custom label when adding URL via LinkPicker

jeryj opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- Link this PR to its associated issue with an appropriate keyword: Closes, See, Follow up to, etc. -->
Closes https://github.com/WordPress/gutenberg/issues/74005

<!-- In a few words, what is the PR actually doing? -->
Fix an issue where selecting a page from the LinkPicker would overwrite a custom link label.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
LinkPicker should respect the exiting label/title if present, not overwrite it.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
Changes the condition in useHandleLinkChange from checking both url and label to only checking if a label exists, ensuring custom labels are always preserved when present.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->
1. Edit a navigation to have an empty link via the code editor view: `<!-- wp:navigation-link {"label":"Empty Link"} /-->`
2. Select this block via the list view (selecting it on canvas opens the link ui popover on canvas)
3. Use the Link Picker in the inspector sidebar to select a new link 
4. The title "Empty Link" should show in the canvas and not be overwritten


### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

## Screenshots or screencast <!-- if applicable -->

<!-- If you would like to upload screenshots, feel free to use the table below when it is useful to show the difference between before and after the change. -->

|Before|After|
|-|-|
|<!-- Before screenshot here -->|<!-- After screenshot here -->|

jeryj committed:
Navigation Link: Preserve custom label when adding URL via LinkPicker

getdave commented on packages/block-library/src/navigation-link/shared/use-handle-link-change.js:
```diff
@@ -42,9 +42,9 @@ export function useHandleLinkChange( { clientId, attributes, setAttributes } ) {
+			// Only include title when there's no existing label
 			// This preserves user-customized labels when updating links
-			if ( ! attributes.url || ! attributes.label ) {
+			if ( ! attributes.label ) {
```
Do we need to check for a length here as well? If the label is empty then replace it?

getdave reviewed (commented):
This works when I test it but the e2e tests are certainly not happy. Let's see what happens.

jeryj committed:
Fix failing block selection in e2e test

jeryj committed:
Also replace label when it's an empty string

jeryj commented:
Thanks for the review - the new test was failing because of a block selection issue, not because of the code. I fixed that and added a check to also replace empty string titles like you suggested. The rest of the e2e test failures look unrelated, so I'm rerunning tests and seeing if they pass.

getdave reviewed (approved):
Works for me. Let's bring it in!
